ninja-core/src/main/java/ninja/ReverseRouter.java

```diff
@@ -0,0 +1,421 @@
+            }
+            
+            this.queries.put(name, safeValue(value, raw));
+            
+            return this;
+        }
+        
+        /**
+         * Add a map containing pairs with replacements for either path or query
+         * string parameters. This method will URL encode the values when building
+         * the final result.  When replacing path parameters it's recommended
+         * to use the <code>path</code> method since it will validate the parameter
+         * exists, while this method would fallback to adding the value as a query
+         * string parameter if the path parameter did not exist.
+         * @param parameterMap The map containing pairs of name and values.
+         * @return A reference to this builder
+         * @see #path(java.lang.String, java.lang.Object)
+         * @see #query(java.lang.String, java.lang.Object) 
+         */
+        public Builder params(Map<String,Object> parameterMap) {
+            return setParams(parameterMap, false);
+        }
+        
+        /**
+         * Identical to <code>params</code> except this method will NOT url
+         * encode the values when building the final result.  When replacing path
+         * parameters it's recommended to use the <code>path</code> method since
+         * it will validate the parameter exists, while this method would fallback
+         * to adding the value as a query string parameter if the path parameter
+         * did not exist.
+         * @param parameterMap The map containing pairs of name and values.
+         * @return A reference to this builder
+         * @see #path(java.lang.String, java.lang.Object)
+         * @see #query(java.lang.String, java.lang.Object) 
+         */
+        public Builder rawParams(Map<String,Object> parameterMap) {
+            return setParams(parameterMap, true);
+        }
+        
+        private Builder setParams(Map<String,Object> parameterMap, boolean raw) {
+            if (parameterMap != null) {
+                parameterMap.forEach((name, value) -> setParam(name, value, raw));
+            }
+            return this;
+        }
+        
+        /**
+         * Add a list containing pairs with replacements for either path or query
+         * string parameters. This method will URL encode the values when building
+         * the final result.  When replacing path parameters it's recommended
+         * to use the <code>path</code> method since it will validate the parameter
+         * exists, while this method would fallback to adding the value as a query
+         * string parameter if the path parameter did not exist.
+         * @param parameterMap The list of pairs of name and values.
+         * @return A reference to this builder
+         * @see #path(java.lang.String, java.lang.Object)
+         * @see #query(java.lang.String, java.lang.Object) 
+         */
+        public Builder params(Object... parameterMap) {
```

Comment:
We could also remove that method...  - what do you think? param and query are already perfect...